security/openconnect/src/opnsense/service/templates/OPNsense/Openconnect/opnsense-openconnect

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+#
+# $FreeBSD$
+#
+# PROVIDE: opnsense-openconnect
+# REQUIRE: SERVERS
+# KEYWORD: shutdown
+#
+
+. /etc/rc.subr
+
+name=openconnect
+
+stop_cmd=openconnect_stop
+start_cmd=openconnect_start
+status_cmd=openconnect_status
+rcvar=openconnect_enable
+
+load_rc_config opnsense-openconnect
+pidfile=/var/run/${name}.pid
+command=/usr/local/sbin/${name}
+
+[ -z "$openconnect_enable" ] && openconnect_enable="NO"
+
+# status of openconnect
+openconnect_status()
+{
+	 if [ -n "$rc_pid" ]; then
+	     echo "${name} is running as pid $rc_pid."
+	     return 0
+	 else
+	     echo "${name} is not running."
+	 fi
+}
+
+# stop openconnect
+openconnect_stop()
+{
+	echo "stopping openconnect"
+	killall openconnect
+	ifconfig ocvpn0 destroy
+	return 0
+}
+
+# start openconnect
+openconnect_start()
+{
+{% if helpers.exists('OPNsense.openconnect.general.enabled') and OPNsense.openconnect.general.enabled == '1' %}
+{%   if helpers.exists('OPNsense.openconnect.general.server') and OPNsense.openconnect.general.server != '' %}
+{%     if helpers.exists('OPNsense.openconnect.general.user') and OPNsense.openconnect.general.user != '' %}
+{%       if helpers.exists('OPNsense.openconnect.general.password') and OPNsense.openconnect.general.password != '' %}  
+        echo "starting openconnect"  
+	echo "{{ OPNsense.openconnect.general.password }}" | /usr/local/sbin/openconnect -u {{ OPNsense.openconnect.general.user }} {{ OPNsense.openconnect.general.server }} --pid-file=/var/run/openconnect.pid -b -q -i tun30000 2>&1 > /dev/null
```

Comment:
* pid file will be overwritten by the next instance
* maybe all of them use the same tun device

Reply:
You mean to fix like in https://github.com/opnsense/plugins/pull/458/files#diff-b42ee4592f185b3cdfa293bdffd10af0 ?

Reply:
basically rc.conf.d and rc.d files work in tandem. you set all your user-facing options in the rc.conf.d file which can be used as a escaping mechanism against RCE as well (single quotes for all injected settings as long as single quotes are disable in the respective config options).

the rc.d file reads the variables and uses them, that makes the script static and easier to maintain. :)